[quant] aten::repeat work for quantized tensor

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40653 [quant][graphmode][fix] remove unsupported ops in the list
* **#40644 [quant] aten::repeat work for quantized tensor**
* #40624 [quant][graphmode][fix] cloning schema in insert_observers

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22268558](https://our.internmc.facebook.com/intern/diff/D22268558)

jerryzh168 committed:
[quant] aten::repeat work for quantized tensor

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on aten/src/ATen/native/TensorShape.cpp:
```diff
@@ -762,7 +762,12 @@ Tensor repeat(const Tensor& self, IntArrayRef repeats) {
+  if (self.is_quantized()) {
+    result = at::empty_quantized(target_size, self);
+  } else {
+    result = at::empty(target_size, self.options());
```
does dispatching on tensor type not work for `at::empty`?

jerryzh168 replied:
yeah, quantized parameters are not included in TensorOptions(self.options())

supriyar commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -1099,6 +1099,11 @@
   variants: method
   device_guard: False
 
+- func: empty_quantized(int[] size, Tensor original) -> Tensor
```
The name seems a little non-intuitive to me. Maybe make it explicit that original has to be a quantized tensor.

jerryzh168 replied:
sg

jerryzh168 committed:
Update on "[quant] aten::repeat work for quantized tensor"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22268558](https://our.internmc.facebook.com/intern/diff/D22268558)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] aten::repeat work for quantized tensor"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22268558](https://our.internmc.facebook.com/intern/diff/D22268558)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] aten::repeat work for quantized tensor"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22268558](https://our.internmc.facebook.com/intern/diff/D22268558)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] aten::repeat work for quantized tensor"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22268558](https://our.internmc.facebook.com/intern/diff/D22268558)

[ghstack-poisoned]
